docs/rfc/0696-p2p-node.adoc

```diff
@@ -0,0 +1,318 @@
+= RFC: Peer-to-peer Node
+:author: @FintanH, @xla
+:revdate: 2021-06-31
+:revremark: draft
+:toc:
+:toc-placement: preamble
+
+* Author: {author}
+* Date: {revdate}
+* Status: {revremark}
+* Community discussion: n/a
+* Tracking Issue: n/a
+
+== Motivation
+
+So far long-running binaries providing a link p2p daemon came in two
+incarnations, the *radicle-upstream* proxy and the *radicle-bins* seed node.
+With the former being the main subject of experimentation of higher-level
+non-core protocols (e.g. requesting new identities on the network) it developed
+into a monolith due to its locality close to the upstream project and a constant
+push to reflect highly specific product needs to drive the UI. While beneficial
+to validate the feasibility of the general ideas of the link protocol it now
+presents the reality that there is no simple way to run the link p2p stack
+without the idiosyncrasies of the application specific proxy. This is a massive
+hindrance for further adoption and recomposition in other contexts and needs to
+be addressed to enable the development of a healthy ecosystem.
+
+== Overview
+
+This RFC expands on the peer-to-peer node outlined in RFC 682 to propose a new
+stand-alone daemon with focus on driving the p2p stack, exposing a minimal API
+and defer supervision to more integrated and mature system services. Furthermore
+it tries to prove that all daemons are made equal and only distinct with regards
+to the configuration given (e.g. tracking behaviour) and lifetime of the process
+(i.e. socket-activated or long-running).
+
+It also strives to recommend APIs and mechanisms - in the form of non-core
+protocols - which remove the need for online requests made by clients to a
+running node.
+
+== Terminology and Conventions
+
+The key words "`MUST`", "`MUST NOT`", "`REQUIRED`", "`SHALL`", "`SHALL NOT`",
+"`SHOULD`", "`SHOULD NOT`", "`RECOMMENDED`", "`NOT RECOMMENDED`", "`MAY`", and
+"`OPTIONAL`" in this document are to be interpreted as described in <<RFC2119>>
+and <<RFC8174>> when, and only when, they appear in all capitals, as shown here.
+
+CBOR <<RFC8949>> datatype definitions are given using the notation devised in
+CDDL <<RFC8610>>. By convention, `struct`-like datatypes are encoded as CBOR
+maps, where the map key is the zero-based numeric index of the field in
+declaration order.
+
+== Protocols
```

Comment:
It might be helpful for future reference to introduce the terminology `Protocol` and `Subroutine` properly here. A `Subroutine` is kind of an event loop which drives a `Protocol`, or?

Reply:
Would you consider 8dfb717 sufficient?